### ISSUE TITLE
Exposing base member from derived, when base is not wrapped

### ISSUE THREAD

**Issue by pschella** [pybind/pybind11#910]
If I have a base class `B` with a data member `x` (of type `float`, but that is irrelevant) and a derived class `D`. And if I now expose only `D` (because `B` for some other reason should not be exposed) as: `py::class_<D>(m, "D").def(py::init<>()).def_readonly("x", &D::x);` Accessing `x` from Python fails with:

```
E TypeError: (): incompatible function arguments. The following argument types are supported:
E 1. (self: B) -> float
E
E Invoked with: <pybind11_tests.D object at 0x108e2e538>

```

This can be fixed with an explicit cast as in `.def_readonly("x", (float D::*) &D::x);`, but that should not really be required (as pointed out my @jagerman on Gitter: Pybind doesn't know anything about the `D : B` relationship here, which is why it happens. It ought to be possible to make it work properly (and `static_assert` fail when the member isn't a superclass member)).

**Comment by bmerry** [comment]
This looks like a duplicate of #854, and I think #855 should also fix it (and in a more general way).

**Comment by dean0x7d** [comment]
Fixed in #855.

**Comment by pschella** [comment]
Thank you!
